src/log/File.cc

```diff
@@ -112,21 +113,11 @@ void
 logfilePrintf(Logfile * lf, const char *fmt,...)
 {
     va_list args;
-    char buf[8192];
-    int s;
-
     va_start(args, fmt);
-
-    s = vsnprintf(buf, 8192, fmt, args);
-
-    if (s > 8192) {
-        s = 8192;
-
-        if (fmt[strlen(fmt) - 1] == '\n')
-            buf[8191] = '\n';
-    }
-
-    logfileWrite(lf, buf, (size_t) s);
+    static SBuf sbuf;
+    sbuf.clear();
+    sbuf.vappendf(fmt, args);
```

Comment:
AFAICT, SBuf::vappendf() is completely broken right now. It does not increase the size in a loop and some of its return value tests do not make sense to me. I would expect it to fail on many logformat specifications, especially a trivial one that logs just the URL (i.e., short format specs but long parameter string).

If I am right, please fix that method so that we can actually reuse it here without breaking everything. You already have already written most of the code that you can resurrect in SBuf::vappendf() to fix that method. Just keep my earlier change requests in mind when doing so.

Reply:
Reading back through the comments above it is not clear to me what changes have been requested for this pull request.